mito-ai/src/tests/AiChat/ChatMessage.test.tsx

```diff
@@ -0,0 +1,355 @@
+    }))
+}) as unknown as IRenderMimeRegistry;
+
+// Create base props for the component
+const createMockProps = (overrides = {}) => ({
+    message: createMockMessage('user', 'Hello, can you help me with pandas?'),
+    messageType: 'openai message' as ChatMessageType,
+    codeCellID: 'test-cell-id',
+    messageIndex: 0,
+    promptType: 'chat' as PromptType,
+    mitoAIConnectionError: false,
+    mitoAIConnectionErrorType: null,
+    notebookTracker: createMockNotebookTracker(),
+    renderMimeRegistry: createMockRenderMimeRegistry(),
+    app: { commands: { execute: jest.fn() } } as unknown as JupyterFrontEnd,
+    isLastAiMessage: false,
+    operatingSystem: 'mac' as OperatingSystem,
+    previewAICode: jest.fn(),
+    acceptAICode: jest.fn(),
+    rejectAICode: jest.fn(),
+    onUpdateMessage: jest.fn(),
+    onDeleteMessage: jest.fn(),
+    contextManager: { getVariables: jest.fn(() => []) } as unknown as IContextManager,
+    codeReviewStatus: 'chatPreview' as CodeReviewStatus,
+    ...overrides
+});
+
+// Helper function to render the component
+const renderChatMessage = (props = {}) => {
+    cleanup();
+    return render(<ChatMessage {...createMockProps(props)} />);
+};
+
+describe('ChatMessage Component', () => {
+    beforeEach(() => {
+        cleanup();
+        jest.clearAllMocks();
+        // Clear previous callbacks
+        (window as any).__chatInputCallbacks = null;
+    });
+
+    afterEach(() => {
+        cleanup();
+    });
+
+    describe('Rendering', () => {
+        it('renders a user message correctly', () => {
+            renderChatMessage({
+                message: createMockMessage('user', 'Hello, can you help me with pandas?')
+            });
+
+            // Since we mocked MarkdownBlock, we can find the text directly
+            expect(screen.getByText('Hello, can you help me with pandas?')).toBeInTheDocument();
+
+            // Check for the user message class
+            const messageElement = screen.getByText('Hello, can you help me with pandas?');
+            const messageContainer = messageElement.closest('.message');
+            expect(messageContainer).toHaveClass('message-user');
+        });
+
+        it('renders an assistant message correctly', () => {
+            renderChatMessage({
+                message: createMockMessage('assistant', 'Yes, I can help with pandas. What do you need?'),
+                messageType: 'openai message'
+            });
+
+            // Check for the assistant message text and class
+            expect(screen.getByText('Yes, I can help with pandas. What do you need?')).toBeInTheDocument();
+            const messageElement = screen.getByText('Yes, I can help with pandas. What do you need?');
+            const messageContainer = messageElement.closest('.message');
+            expect(messageContainer).toHaveClass('message-assistant-chat');
+        });
+
+        it('renders a code block in an assistant message', () => {
```

Comment:
For this one, can we check that we have a code block along with like the Copy and Overwite buttons?